test/test_view_ops.py

```diff
@@ -457,6 +463,61 @@ def test_reshape_nonview(self, device):
         nv[6] = 0
         self.assertNotEqual(t[1, 1], nv[6])
 
+    def test_flatten_view(self, device):
+        def assert_is_view(t, v, t_is_view=False):
+            idx_t = (0,) * t.ndim
+            idx_v = (0,) * v.ndim
+            if not t_is_view:
+                self.assertTrue(self.is_view_of(t, v))
+            else:
+                self.assertTrue(self.is_view_of_same_base(t, v))
+            v[idx_v] = 0
+            self.assertEqual(t[idx_t], v[idx_v])
+
+        t = torch.ones(1, 2, 3, 4, device=device)
+        v = t.flatten()
+        assert_is_view(t, v)
+
+        # zero-dimensional tensor
+        t = torch.tensor(1, device=device)
+        v = t.flatten()
+        assert_is_view(t, v)
+
+        t = torch.ones(1, 2, 3, 4, device=device).transpose(2, 3)
+        v = t.flatten(0, 1)
+        assert_is_view(t, v, True)
+
+        # stride[i] = stride[i + 1] * size[i + 1] is satisfied for 3 groups:
+        t = torch.ones(720, device=device) \
+            .as_strided((2, 3, 2, 3, 5, 4), (6, 2, 15, 5, 1, 0))
+        #               [--1--|---2---|-3-] [--1--|----2---|-3-]
+        v1 = t.flatten(0, 1)
+        v2 = v1.flatten(1, 3)
+        v3 = v2.flatten(2, 2)
+        assert_is_view(t, v1, True)
+        assert_is_view(t, v2, True)
+        assert_is_view(t, v3, True)
+
+    def test_flatten_nonview(self, device):
+        def assert_is_nonview(t, nv):
```

Comment:
This function can stay as-is because it always asserts that nv is not a view of t.